packages/calcite-components/src/components/block/block.tsx

```diff
@@ -209,9 +243,22 @@ export class Block
 
   /**
    * Emits when the component's header is clicked.
+   *
+   * @deprecated use `openClose` events instead.
    */
   @Event({ cancelable: false }) calciteBlockToggle: EventEmitter<void>;
 
+  /** Fires when the component is requested to be closed and before the closing transition begins. */
+  @Event({ cancelable: false }) calciteBlockBeforeClose: EventEmitter<void>;
+
+  /** Fires when the component is closed and animation is complete. */
+  @Event({ cancelable: false }) calciteBlockClose: EventEmitter<void>;
+
+  /** Fires when the component is added to the DOM but not rendered, and before the opening transition begins. */
+  @Event({ cancelable: false }) calciteBlockBeforeOpen: EventEmitter<void>;
```

Comment:
Nitpick: can the event emitters follow alphabetical order?

Applies to other added emitters.